src/pydata_sphinx_theme/assets/scripts/pydata-sphinx-theme.js

```diff
@@ -212,18 +212,26 @@ var toggleSearchField = () => {
 
   // if the input field is the hidden one (the one associated with the
   // search button) then toggle the button state (to show/hide the field)
-  let searchPopupWrapper = document.querySelector(".search-button__wrapper");
-  let hiddenInput = searchPopupWrapper.querySelector("input");
+  let searchDialog = document.querySelector(".search-button__search-container");
```

Comment:
You do point later on that there might  be multiple search dialog, how does this interact with this fact, aren't we making the supposition that there is only one per page here ? 

I guess there is a single `.search-button__search-container` (so maybe this one should be an id ? 

Just checking I might just misundeerstand.

Reply:
There should be one and only one search **dialog modal**, but there can be multiple search input fields. I would actually like to maybe start a discussion around this because I think the currently allowed behavior is a bit confusing.

I'm +1 for changing to an id. I'm thinking... `#pst-search-dialog` or is there some naming convention I need to follow here?

Reply:
And here's the discussion:

- https://github.com/pydata/pydata-sphinx-theme/discussions/1935